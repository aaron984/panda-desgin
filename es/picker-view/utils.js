export var setTransform = function setTransform(nodeStyle, value) {
  nodeStyle.transform = value;
  nodeStyle.webkitTransform = value;
};
export var setTransition = function setTransition(nodeStyle, value) {
  nodeStyle.transition = value;
  nodeStyle.webkitTransition = value;
};
export var Velocity = function () {
  var minInterval = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 30;
  var maxInterval = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 100;
  var time = 0;
  var initY = 0;
  var velocity = 0;
  var recorder = {
    record: function record(y) {
      var now = +new Date();
      velocity = (y - initY) / (now - time);

      if (now - time >= minInterval) {
        velocity = now - time <= maxInterval ? velocity : 0;
        initY = y;
        time = now;
      }
    },
    getVelocity: function getVelocity(y) {
      if (y !== initY) {
        recorder.record(y);
      }

      return velocity;
    }
  };
  return recorder;
}();