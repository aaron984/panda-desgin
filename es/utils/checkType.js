export var isObject = function isObject(value) {
  return Object.prototype.toString.call(value) === '[object Object]';
};
export var isString = function isString(value) {
  return Object.prototype.toString.call(value) === '[object String]';
};
export var isNull = function isNull(value) {
  return Object.prototype.toString.call(value) === '[object Null]';
};
export var isFunction = function isFunction(value) {
  return Object.prototype.toString.call(value) === '[object Function]';
};
export var isArray = function isArray(value) {
  return Object.prototype.toString.call(value) === '[object Array]';
};