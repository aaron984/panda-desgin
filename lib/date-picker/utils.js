"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = formatFn;

function formatIt(date, form) {
  var pad = function pad(n) {
    return n < 10 ? "0".concat(n) : n;
  };

  var dateStr = "".concat(date.getFullYear(), "-").concat(pad(date.getMonth() + 1), "-").concat(pad(date.getDate()));
  var timeStr = "".concat(pad(date.getHours()), ":").concat(pad(date.getMinutes()));

  if (form === 'YYYY-MM-DD') {
    return dateStr;
  }

  if (form === 'HH:mm') {
    return timeStr;
  }

  return "".concat(dateStr, " ").concat(timeStr);
}

function formatFn(instance, value) {
  var formatsEnum = {
    date: 'YYYY-MM-DD',
    time: 'HH:mm',
    datetime: 'YYYY-MM-DD HH:mm'
  };
  var format = instance.props.format;
  var type = typeof format;

  if (type === 'string') {
    return formatIt(value, format);
  }

  if (type === 'function') {
    return format(value);
  }

  return formatIt(value, formatsEnum[instance.props.mode]);
}