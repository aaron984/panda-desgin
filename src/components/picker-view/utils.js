export const setTransform = (nodeStyle, value) => {
  nodeStyle.transform = value;
  nodeStyle.webkitTransform = value;
};

export const setTransition = (nodeStyle, value) => {
  nodeStyle.transition = value;
  nodeStyle.webkitTransition = value;
};

export const Velocity = ((minInterval = 30, maxInterval = 100) => {
  let time = 0;
  let initY = 0;
  let velocity = 0;

  const recorder = {
    record: (y) => {
      const now = +new Date();
      velocity = (y - initY) / (now - time);

      if (now - time >= minInterval) {
        velocity = now - time <= maxInterval ? velocity : 0;
        initY = y;
        time = now;
      }
    },
    getVelocity: (y) => {
      if (y !== initY) {
        recorder.record(y);
      }
      return velocity;
    },
  };
  return recorder;
})();
