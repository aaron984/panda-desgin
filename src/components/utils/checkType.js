export const isObject = value => Object.prototype.toString.call(value) === '[object Object]';
export const isString = value => Object.prototype.toString.call(value) === '[object String]';
export const isNull = value => Object.prototype.toString.call(value) === '[object Null]';
export const isFunction = value => Object.prototype.toString.call(value) === '[object Function]';
export const isArray = value => Object.prototype.toString.call(value) === '[object Array]';
