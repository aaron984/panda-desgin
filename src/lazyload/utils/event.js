export function on(el, eventName, callback, opts = false) {
  el.addEventListener(eventName, callback, opts);
}

export function off(el, eventName, callback, opts = false) {
  opts = opts || false;
  el.removeEventListener(eventName, callback, opts);
}
