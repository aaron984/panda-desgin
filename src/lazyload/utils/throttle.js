/*eslint-disable */
export default function throttle(fn, threshhold = 250, scope) {
  let last;
  let deferTimer;

  return function () {
    let context = scope || this;

    let now = +new Date;
    let args = arguments;
    if (last && now < last + threshhold) {
      clearTimeout(deferTimer);
      deferTimer = setTimeout(function () {
        last = now;
        fn.apply(context, args);
      }, threshhold);
    } else {
      last = now;
      fn.apply(context, args);
    }
  };
}
