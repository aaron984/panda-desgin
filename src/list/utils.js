export default {
  getScrollEventTarget(element, rootParent = window) {
    let currentNode = element;
    while (currentNode && currentNode.tagName !== 'HTML' && currentNode.tagName !== 'BODY' && currentNode.nodeType === 1 && currentNode !== rootParent) {
      const overflowY = this.getComputedStyle(currentNode).overflowY;
      if (overflowY === 'scroll' || overflowY === 'auto') {
        return currentNode;
      }
      currentNode = currentNode.parentNode;
    }
    return rootParent;
  },

  getScrollTop(element) {
    return 'scrollTop' in element ? element.scrollTop : element.pageYOffset;
  },

  getElementTop(element) {
    return (element === window ? 0 : element.getBoundingClientRect().top) + this.getScrollTop(window);
  },

  getVisibleHeight(element) {
    return element === window ? element.innerHeight : element.getBoundingClientRect().height;
  },

  getComputedStyle: document.defaultView.getComputedStyle.bind(document.defaultView),
};
