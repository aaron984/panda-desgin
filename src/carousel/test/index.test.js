import React from 'react';
import { render, mount } from 'enzyme';
import { triggerDrag } from 'test/utils';
import Carousel from '../index';

describe('Carousel', () => {
  beforeEach(() => {
    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.useRealTimers();
  });

  it('Carousel renders correctly', () => {
    const wrapper = render(
      <Carousel autoplay>
        <div key="1" style={{ width: 100, height: 200 }}>1</div>
        <div key="2" style={{ width: 100, height: 200 }}>2</div>
      </Carousel>);
    expect(wrapper).toMatchSnapshot();
  });

  it('Carousel renders correctly', () => {
    const wrapper = mount(
      <Carousel autoplay={100}>
        <div key="1" style={{ width: 100, height: 200 }}>1</div>
        <div key="2" style={{ width: 100, height: 200 }}>2</div>
      </Carousel>);

    const instance = wrapper.instance();
    expect(instance.state.active).toEqual(0);
    jest.runOnlyPendingTimers();
    jest.runOnlyPendingTimers();
    expect(instance.state.active).toEqual(1);
  });

  it('Carousel renders correctly', () => {
    const wrapper = mount(
      <Carousel >
        <div key="1" style={{ width: 100, height: 200 }}>1</div>
        <div key="2" style={{ width: 100, height: 200 }}>2</div>
        <div key="3" style={{ width: 100, height: 200 }}>2</div>
      </Carousel>);

    const instance = wrapper.instance();
    expect(instance.state.active).toEqual(0);
    triggerDrag(instance.el, -100, 0);
    expect(instance.state.active).toEqual(1);
    triggerDrag(instance.el, -100, 0);
  });

  it('Carousel renders correctly', () => {
    const wrapper = mount(
      <Carousel vertical loop={false}>
        <div key="1" style={{ width: 100, height: 200 }}>1</div>
        <div key="2" style={{ width: 100, height: 200 }}>2</div>
      </Carousel>);

    const instance = wrapper.instance();
    instance.resetTouchStatus();

    expect(instance.state.active).toEqual(0);
    triggerDrag(instance.el, 0, -100);
    expect(instance.state.active).toEqual(1);
  });

  it('Carousel renders correctly', () => {
    const wrapper = mount(
      <Carousel touchable={false} initialSwipe={3}>
        <div key="1" style={{ width: 100, height: 200 }}>1</div>
        <div key="2" style={{ width: 100, height: 200 }}>2</div>
        <div key="2" style={{ width: 100, height: 200 }}>2</div>
        <div key="2" style={{ width: 100, height: 200 }}>2</div>
      </Carousel>);

    const instance = wrapper.instance();
    triggerDrag(instance.el, 0, -100);
    expect(instance.state.active).toEqual(3);
    triggerDrag(instance.el, 0, -100);
    expect(instance.state.active).toEqual(3);

    instance.active = -1;
    instance.correctPosition();
    expect(instance.state.active).toEqual(3);
  });
});
