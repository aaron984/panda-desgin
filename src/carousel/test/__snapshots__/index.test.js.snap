// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel Carousel renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "panda-swipe",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "panda-swipe__track",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "panda-swipe-item",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "style": "width:100px;height:200px",
                },
                "children": Array [
                  Object {
                    "data": "1",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "style": undefined,
                },
                "x-attribsPrefix": Object {
                  "style": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "panda-swipe-item",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "style": "width:100px;height:200px",
                  },
                  "children": Array [
                    Object {
                      "data": "2",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "style": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "style": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "panda-swipe-item",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "style": "width:100px;height:200px",
                },
                "children": Array [
                  Object {
                    "data": "2",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "style": undefined,
                },
                "x-attribsPrefix": Object {
                  "style": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "panda-swipe-item",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "style": "width:100px;height:200px",
                  },
                  "children": Array [
                    Object {
                      "data": "1",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "style": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "style": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "panda-swipe__indicators",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "panda-swipe__indicator panda-swipe__indicator--active",
              },
              "children": Array [],
              "name": "i",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "panda-swipe__indicator",
                },
                "children": Array [],
                "name": "i",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "panda-swipe__indicator",
              },
              "children": Array [],
              "name": "i",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "panda-swipe__indicator panda-swipe__indicator--active",
                },
                "children": Array [],
                "name": "i",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "panda-swipe__indicators",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "panda-swipe__indicator panda-swipe__indicator--active",
            },
            "children": Array [],
            "name": "i",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "panda-swipe__indicator",
              },
              "children": Array [],
              "name": "i",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "panda-swipe__indicator",
            },
            "children": Array [],
            "name": "i",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "panda-swipe__indicator panda-swipe__indicator--active",
              },
              "children": Array [],
              "name": "i",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "panda-swipe__track",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "panda-swipe-item",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "style": "width:100px;height:200px",
                  },
                  "children": Array [
                    Object {
                      "data": "1",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "style": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "style": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "panda-swipe-item",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "style": "width:100px;height:200px",
                    },
                    "children": Array [
                      Object {
                        "data": "2",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "style": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "style": undefined,
                    },
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "panda-swipe-item",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "style": "width:100px;height:200px",
                  },
                  "children": Array [
                    Object {
                      "data": "2",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "style": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "style": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "panda-swipe-item",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "style": "width:100px;height:200px",
                    },
                    "children": Array [
                      Object {
                        "data": "1",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "style": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "style": undefined,
                    },
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
